### PR TITLE
[#31] Model dependency capturing

### LINKED ISSUE #31: Automatic container changes capture
We need a capability to automatically capture changes introduced by pymodel training script into a model image. 

At the moment both model training scripts and HTTP server run inside a clean container built from base image. It has two disadvantages:
- Every time a new ML library or just another python package is needed by model, the base docker image must be released. That is time consuming and not viable for a prod environment. 
- If model can not create produce some intermediate data/files to be used inside serving environment.

### PR DESCRIPTION
**Pull request by kirillmakhonin**
This closes #31, #196

### CODE REVIEW

**Comment by kirillmakhonin** [comment]
ok to test

**Comment by kirillmakhonin** [comment]
test this please

File: legion/legion/k8s/utils.py [hunk 1]
```diff
@@ -176,9 +176,10 @@ def get_meta_from_docker_labels(labels):
     )
 
     compatible_labels = {
-        normalize_name(k): normalize_name(v)
```
[the remark this answers is not in the document]
Comment (kirillmakhonin) [reply legion/legion/k8s/utils.py]: @akharlamov this is because of the fact kubernetes allows to setup at max 63 chars strings as label

**Comment by alexander-semenets** [comment]
there is py-gunicorn installed to edi, do we still need it there?

File: legion/legion/containers/docker.py [hunk 2]
```diff
@@ -44,103 +42,136 @@ def build_docker_client(args=None):
     return client
 
 
-def build_docker_image(client, base_image, model_id, model_file, labels,
-                       python_package, python_package_version, python_repository,
-                       docker_image_tag, serving):
+def get_docker_container_id_from_cgroup_line(line):
```
Comment (alexander-semenets) [inline legion/legion/containers/docker.py]: why do we want to split it to the separate function?

**Comment by kirillmakhonin** [comment]
test this please

**Comment by kirillmakhonin** [comment]
@alexander-semenets py-gunicorn has been removed

File: legion/legion/containers/docker.py (hunk 2 above)
Comment (kirillmakhonin) [reply legion/legion/containers/docker.py]: I've just split reading and processing data

**Pull request merged**